codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -36,9 +39,24 @@
 @ChannelHandler.Sharable
 public final class MqttEncoder extends MessageToMessageEncoder<MqttMessage> {
 
-    public static final MqttEncoder INSTANCE = new MqttEncoder();
+    public static final MqttEncoder INSTANCE = new MqttEncoder(null);
 
-    private MqttEncoder() { }
+    private AtomicReference<MqttVersion> mqttVersionRef;
```

Comment:
This will not work as `MqttEncoder` is marked as `@Sharable` and so may be shared between different `Channel` instances.

Reply:
Changed version handling approach, removed atomic references.